packages/components/src/components/Combobox/Combobox.stories.tsx

```diff
@@ -0,0 +1,243 @@
+    description: {
+      story:
+        "The combobox will call the `onChangeInput` function when the input value changes.",
+    },
+  },
+};
+
+export const WithOnSelect = Template.bind({});
+WithOnSelect.args = {
+  ...Default.args,
+  // eslint-disable-next-line no-console
+  onSelectItem: (event) => console.log(event),
+};
+WithOnSelect.parameters = {
+  docs: {
+    description: {
+      story:
+        "The combobox will call the `onSelectItem` function when an option is selected.",
+    },
+  },
+};
+
+export const WithLongItem = (): React.JSX.Element => {
+  const longItem = {
+    value:
+      "Option Five which is a really long item and should overflow to the next line",
+    label:
+      "Option Five which is a really long item and should overflow to the next line",
+  };
+
+  return (
+    <Box.div w="250px">
+      <Label htmlFor="select-option">Select an option</Label>
+      <Combobox id="select-option" items={[...items, longItem]} name="select" />
+    </Box.div>
+  );
+};
+
+type CustomItem = {
+  value: string;
+  label: string;
+  disabled?: boolean;
+};
+
+export const WithCustomFilterItem = (): React.JSX.Element => {
+  const customItems: CustomItem[] = [
+    ...items,
+    { value: "New option 1", label: "New option 1", disabled: true },
+    { value: "New option 2", label: "New option 2", disabled: true },
+  ];
+
+  const customFilter = (inputValue: string, item: CustomItem): boolean =>
+    includes(toLower(item.label), toLower(inputValue)) && !item.disabled;
+
+  return <Combobox filterItem={customFilter} items={customItems} />;
+};
+WithCustomFilterItem.parameters = {
+  docs: {
+    description: {
+      story:
+        "The combobox will filter the items based on a given function " +
+        "that filters the items based on the input value and the item.",
+    },
+  },
+};
+
+export const WithReactHookForm = (): React.JSX.Element => {
+  interface FormInputs {
+    item: string;
+  }
+
+  const { control, handleSubmit, clearErrors, formState } = useForm({
+    defaultValues: {
+      item: "",
+    },
+  });
+
+  useEffect(() => {
+    clearErrors();
+  }, [clearErrors, formState.isDirty]);
+
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  const onSubmit: SubmitHandler<FormInputs> = (data) =>
+    alert(JSON.stringify(data, null, 2));
+
+  return (
+    <Box.form noValidate onSubmit={handleSubmit(onSubmit)}>
+      <Controller
+        control={control}
+        name="item"
+        render={({ field, fieldState }) => (
+          <>
+            <Label htmlFor="select-option" required>
+              Select an option
+            </Label>
+            <Combobox
+              {...field}
+              aria-describedby="select-option-help-text"
+              hasError={fieldState.invalid}
+              id="select-option"
+              items={items}
+              onChangeInput={(event) => field.onChange(event.target.value)}
+              onSelectItem={(value) => field.onChange(value)}
```

Comment:
when selecting an item the label is replaced with the value of the option. Example:
1) type "two"
2) select option "Option Two"
3) input now shows "2" instead of "Option Two"